### PR TITLE
Bug 1741813 - Make setUploadEnabled documentation more robust

### PR DESCRIPTION
**Pull request by brizental**
[doc only]

I tried to add some more interesting examples, but the only one I am confident in is the JS one. The Swift and Kotlin ones I copied and edited from our sample apps and the other ones I just didn't. Let me know if the ones I put there are not too confusing.

### CODE REVIEW

File: docs/user/reference/general/toggling-upload-status.md [hunk 1]
```diff
@@ -1,62 +1,134 @@
...
+1. Submit a [`deletion-request`](../../user/pings/deletion_request.md) ping.
+2. Cancel scheduled ping uploads.
+3. Clear metrics and pings data from the client, except for the
+  [`first_run_date` and `first_run_hour`](../../user/pings/index.html#the-client_info-section) metrics.
+
+While upload is disabled, metrics aren't recorded and no data is uploaded.
+
+## Enabling upload
+
+When upload is enabled, the Glean SDK will re-initialize it's [core metrics](../../user/collected-metrics/metrics.md).
```
Comment (badboy) [inline docs/user/reference/general/toggling-upload-status.md:22]:
```suggestion
When upload is enabled, the Glean SDK will re-initialize its [core metrics](../../user/collected-metrics/metrics.md).
```

File: docs/user/reference/general/toggling-upload-status.md [hunk 2]
```diff
@@ -1,62 +1,134 @@
...
+        // ...
 
-> If called before `Glean.initialize()` the call to `Glean.setUploadEnabled()` will be ignored.
-> Set the initial state using `uploadEnabled` on `Glean.initialize()`.
+        uploadSwitch.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                Glean.setUploadEnabled(true)
+            } else {
+                Glean.setUploadEnabled(false)
+            }
```
Comment (badboy) [inline docs/user/reference/general/toggling-upload-status.md:54]:
```suggestion
            Glean.setUploadEnabled(isChecked)
```

File: docs/user/reference/general/toggling-upload-status.md [hunk 3]
```diff
@@ -1,62 +1,134 @@
...
 
-> If called before `Glean.initialize()` the call to `Glean.setUploadEnabled()` will be ignored.
-> Set the initial state using `uploadEnabled` on `Glean.initialize()`.
+const uploadSwitch = document.querySelector("input[type=checkbox].upload-switch");
+uploadSwitch.addEventListener("change", event => {
+    if (event.target.checked) {
+        Glean.setUploadEnabled(true);
+    } else {
+        Glean.setUploadEnabled(false);
+    }
```
Comment (badboy) [inline docs/user/reference/general/toggling-upload-status.md:120]: same here I guess (though I do understand that explicitly having both if branches might be easier to read.

```suggestion
    Glean.setUploadEnabled(event.target.checked);
```

File: docs/user/reference/general/toggling-upload-status.md [hunk 4]
```diff
@@ -1,62 +1,134 @@
 # Toggling upload status
 
+The Glean SDKs provide an API for toggling Glean's upload status after initialization.
+
+Applications instrumented with Glean
+[are expected to](../../adding-glean-to-your-project/index.md#glean-integration-checklist)
```
Comment (badboy) [inline docs/user/reference/general/toggling-upload-status.md:6]:
```suggestion
[are expected to](../../user/adding-glean-to-your-project/index.md#glean-integration-checklist)
```

**Review by badboy — APPROVED** [review]
That's a good start. The examples look valid.
Minimal nits and a link fix.

File: docs/user/reference/general/toggling-upload-status.md (hunk 3 above)
Comment (brizental) [reply docs/user/reference/general/toggling-upload-status.md:120]: I think since this is just the docs I'd rather leave both branches.

**Pull request merged**